### ISSUE TITLE
too low accuracy in the example of mnist_cnn

### ISSUE THREAD

**Issue by kingaza** [fchollet/keras#4404]
Hello,

I run the example of mnist_cnn by CPU on my win7. The result is very bad, with fixed test accuracy of 0.098 even after 12 epochs. 
I am new in keras, and want to know how to debug it.
Thx for any suggestion.

The packages have been updated to latest version:
Keras version: 1.1.1
Theano version: 0.8.2

Here is the log
Train on 60000 samples, validate on 10000 samples
Epoch 1/12
60000/60000 [==============================] - 5584s - loss: nan - acc: 0.0988 - val_loss: nan - val_acc: 0.0980
Epoch 2/12
60000/60000 [==============================] - 5578s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 3/12
60000/60000 [==============================] - 5577s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 4/12
60000/60000 [==============================] - 5556s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 5/12
60000/60000 [==============================] - 5559s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 6/12
60000/60000 [==============================] - 5549s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 7/12
60000/60000 [==============================] - 5550s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 8/12
60000/60000 [==============================] - 5546s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 9/12
60000/60000 [==============================] - 5548s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 10/12
60000/60000 [==============================] - 5550s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 11/12
60000/60000 [==============================] - 5549s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Epoch 12/12
60000/60000 [==============================] - 5565s - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
Test score: nan
Test accuracy: 0.098

**Comment by keunwoochoi** [comment]
loss is nan, so there are some errors on the data or so. full code?

**Comment by kingaza** [comment]
just the example of mnist_cnn, without any modification.

**Comment by keunwoochoi** [comment]
really? that's strange. I just ran it, keras version is 1.1.0 and it works well. could you check it out on 1.1.0?

**Comment by kingaza** [comment]
I try again, with the same situation.
But cifar10_cnn seems working well:
Epoch 1/200
  800/50000 [..............................] - ETA: 18821s - loss: 2.3023 - acc: 0.1037

**Comment by kingaza** [comment]
@keunwoochoi
which version of python do you use, 2 or 3?
There is a note in cifar10_cnn:
"Note: the data was pickled with Python 2, and some encoding issues might prevent you
from loading it in Python 3. You might have to load it in Python 2,
save it in a different format, load it in Python 3 and repickle it."

**Comment by keunwoochoi** [comment]
Oh, I'm using python2.7. That could be the reason.

**Comment by kingaza** [comment]
same situation in python 2.7
and i check the train and test data in python 3.5, which are all right. So it is not the problem of pickle data parsing.
Still asking for help~~~

**Comment by kingaza** [comment]
no problem any more, since I move to tensorflow as backend.

**Comment by nouiz** [comment]
You should use Theano dev version with keras, not Theano 0.8.2. We released
Theano 0.9rc1 this week.
